common/pom.xml

```diff
@@ -37,11 +37,21 @@
       <artifactId>drill-protocol</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-engine</artifactId>
+      <version>${junit.version}</version>
```

Comment:
Shouldn't we use dependencyManagement for this? (It seems the project uses it sometimes but not always)?
Looks like JUnit project publishes a BOM artifact (org.junit:junit-bom) to make sure that all JUnit dependencies are on the same version.

Also, shouldn't this dependencies being provided as test dependencies (since no main code seems to reference junit5 api). In which case, I guess they could even be dropped since they are declared with the right scope in the top level pom.xml this module is a child of?

Reply:
@laurentgo Very good suggestions!
I've pushed a new commit that uses the junit-bom in dependencyManagement, and uses junit-jupiter instead of junit-jupiter-engine. This way junit-jupiter-api, junit-jupiter-engine and junit-jupiter-params are included as transitive dependencies and one could write use the full power of JUnit 5.x when writing tests.